src/test/java/com/github/checkstyle/regression/git/DiffParserTest.java

```diff
@@ -64,4 +72,18 @@ public void testParse() throws Exception {
         assertEquals(1, changes.size());
         assertEquals("HelloWorld", changes.iterator().next().getPath());
     }
+
+    @Test
+    public void testParseDeleteChange() throws Exception {
+        try (Repository repository = GitUtils.createNewRepository()) {
+            repositories.add(repository);
```

Comment:
I am not expecting you to have to manually add the repository to the list.
`GitUtils.createNewRepository` should do this for you. You should also be calling a method in `GitUtils` to clear it.
The field should be private in it.